File: two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const indexed_nums = nums.map((n, i) => ({n, i}))
    for (const { n: num1, i } of indexed_nums) {
        for (const { n: suposed_num2, i: j } of indexed_nums) {
            if (j !== i) {
                if ((num1 + suposed_num2) === target) {
                    return [i, j]
                }
            }
        }
    }
};

File: two-sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {

        let mut end_vec = vec![0 as i32, 1 as i32];
        for (i, num1) in nums.iter().enumerate() {
            for (j, suposed_num2) in nums.iter().enumerate() {
                if (i != j) {
                    if (num1 + suposed_num2 == target) {
                        end_vec = vec![i as i32, j as i32].to_vec();
                    }
                }
            };
        };

        return end_vec
    }
}